lib/Data/Database.php

```diff
@@ -542,7 +548,16 @@ protected function _getExpiredPastes($batchsize)
     private static function _exec($sql, array $params)
     {
         $statement = self::$_db->prepare($sql);
-        $result    = $statement->execute($params);
+        foreach ($params as $key => &$parameter) {
+            if (is_int($parameter)) {
+                $statement->bindParam($key + 1, $parameter, PDO::PARAM_INT);
```

Comment:
Unfortunately, since bindParam only evaluates everything in the end, passing a variable within the loop and then execute outside would drop the variable and make the position null (or so I think, because it was originally coded as a variable, but it fails to function)...

Reply:
Hm hardly can think of that – that would basically mean it influences the whole syntax/scope of variables in PHP. I doubt so, so I'd try it, but good point to keep in mind in case it fails :astonished:

Reply:
We had one inside the loop, but a recent revision dropped it again. I re-added it.